### PR TITLE
[PREVIEW] Sscs 3899 logout

### PR DESCRIPTION
**Pull request by chrisgrimble**
Looks like logout was working but we did want to refactor the login controller a bit.

Changed login link to /sign-in
Redirect / to /sign-in
Merged dummy-login and login so the shared dependencies do not need to be exposed.
Get idam to redirect to /sign-in rather than /redirect/oauth as this can end up in emails
and they were starting to do the same thing.

### CODE REVIEW

File: app/server/controllers/login.ts [hunk 1]
```diff
@@ -69,6 +73,23 @@ function getIdamCallback(
...
+function getDummyLogin(req: Request, res: Response) {
+  return res.render('dummy-login.html');
+}
+
+function postDummyLogin(getOnlineHearing) {
+  return async(req: Request, res: Response, next: NextFunction) => {
+    const email: string = req.body['username'];
+
+    try {
+      return loadHearingAndEnterService(getOnlineHearing, email, req, res);
```
Comment (davet1985) [inline app/server/controllers/login.ts]: I think we might need to `await` this, as it contains promises.

File: app/server/controllers/login.ts [hunk 2]
```diff
@@ -14,8 +13,24 @@ import * as superAgent from 'superagent';
...
+}
+
+function getLogout(req: Request, res: Response) {
+  const sessionId: string = req.session.id;
+  req.session.destroy(error => {
+    if (error) {
+      logger.error(`Error destroying session ${sessionId}`);
+    }
+    logger.info(`Session destroyed ${sessionId}`);
+    return res.redirect(Paths.login);
```
Comment (davet1985) [inline app/server/controllers/login.ts]: Probably one for later, but we should probably be destroying the session in IDAM too, but not sure what that involves.

File: app/server/controllers/login.ts [hunk 3]
```diff
@@ -80,26 +101,32 @@ async function loadHearingAndEnterService(
 
     return res.render('email-not-found.html');
   }
-  const hearing: OnlineHearing = response.body;
-  req.session.hearing = hearing;
+  req.session.hearing = response.body;
```
Comment (davet1985) [inline app/server/controllers/login.ts]: Can you add the `OnlineHearing` type here.

File: app/server/controllers/login.ts (hunk 1 above)
Comment (chrisgrimble) [reply app/server/controllers/login.ts]: done

**Pull request merged**